java/tools/src/main/java/org/apache/tsfile/tools/TsFileTool.java

```diff
@@ -0,0 +1,498 @@
+              CompressionType.UNCOMPRESSED));
+      columnTypes.add(Tablet.ColumnType.MEASUREMENT);
+    }
+    return new TableSchema(tableName, measurementSchemas, columnTypes);
+  }
+
+  private static boolean writeTsFile(String fileName, List<String> lineList) {
+    final File tsFile = new File(outputDirectoryStr, fileName);
+    TsFileWriter writer = null;
+    try {
+      writer = new TsFileWriter(tsFile);
+      writer.setGenerateTableSchema(true);
+      Map<String, Object> defaultMap = new HashMap<>();
+      TableSchema tableSchema =
+          genTableSchema(schema.idColumns, schema.csvColumns, schema.tableName, defaultMap);
+      writer.registerTableSchema(tableSchema);
+      Tablet tablet = genTablet(tableSchema, lineList, defaultMap);
+      if (tablet != null) {
+        writer.writeTable(tablet);
+        return true;
+      } else {
+        return false;
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
+      LOGGER.error("Failed to write file: " + tsFile);
+      return false;
+    } finally {
+      if (writer != null) {
+        try {
+          writer.close();
+        } catch (IOException e) {
+          e.printStackTrace();
+        }
+      }
+    }
+  }
+
+  private static Tablet genTablet(
+      TableSchema tableSchema, List<String> lineList, Map<String, Object> defaultMap) {
+    int num = lineList.size();
+    Tablet tablet =
+        new Tablet(
+            tableSchema.getTableName(),
+            tableSchema.getColumnSchemas(),
+            tableSchema.getColumnTypes(),
+            num);
+
+    Map<String, Integer> map = new HashMap<>();
+    for (int i = 0; i < schema.csvColumns.size(); i++) {
+      SchemaParser.Column column = schema.csvColumns.get(i);
+      map.put(column.name, i);
+    }
+    try {
+      for (int i = 0; i < num; i++) {
+        String line = lineList.get(i);
+        String[] lineArray = line.split(schema.separator);
+        long timestamp =
+            DateTimeUtils.convertTimestampOrDatetimeStrToLongWithDefaultZone(
+                lineArray[schema.timeColumnIndex], schema.timePrecision);
+
+        tablet.addTimestamp(i, timestamp);
+        List<IMeasurementSchema> columnSchemas = tableSchema.getColumnSchemas();
+        for (int j = 0; j < columnSchemas.size(); j++) {
+          IMeasurementSchema columnSchema = columnSchemas.get(j);
+          if (defaultMap.get(columnSchema.getMeasurementId()) != null) {
+            tablet.addValue(
+                columnSchema.getMeasurementId(),
+                i,
+                defaultMap.get(columnSchema.getMeasurementId()));
+          } else {
+            String value = lineArray[map.get(columnSchema.getMeasurementId())];
+            if (value.equals(schema.nullFormat)) {
+              value = null;
+            }
+            tablet.addValue(
+                columnSchema.getMeasurementId(),
+                i,
+                getValue(columnSchema.getType(), value, tableSchema.getColumnTypes().get(j)));
```

Comment:
No need to call getValue when it is null.
```suggestion
            String valueString = lineSplits[columnIndexMap.get(columnSchema.getMeasurementId())];
            Object value = null
            if (!valueString.equals(schema.nullFormat)) {
              value = getValue(columnSchema.getType(), value, tableSchema.getColumnTypes().get(j));
            }
            
            tablet.addValue(
                columnSchema.getMeasurementId(),
                i,
                value);
```